packages/eas-build-job/src/step.ts

```diff
@@ -126,3 +144,16 @@ export const StepZ = z.union([ShellStepZ, FunctionStepZ]);
  *  - run: echo Hello, world!
  */
 export type Step = z.infer<typeof StepZ>;
+
+export function validateSteps(maybeSteps: unknown): Step[] {
+  const steps = z.array(StepZ).min(1).parse(maybeSteps);
+  return steps;
+}
+
+export function isStepShellStep(step: Step): step is ShellStep {
```

Comment:
Hmm, is there a property that differentiates between shell and function step? It seems wrong to parse steps in the beginning and later parse them again for type differentiation.

Reply:
Yeah, there is. I will change it 🙌